packages/grafana-data/src/types/events.ts

```diff
@@ -0,0 +1,181 @@
+
+export const locationChange = eventFactory<LocationChangePayload>('location-change');
+
+export const timepickerOpen = eventFactory('timepickerOpen');
+export const timepickerClosed = eventFactory('timepickerClosed');
+
+export const showModal = eventFactory<ShowModalPayload>('show-modal');
+export const showConfirmModal = eventFactory<ShowConfirmModalPayload>('confirm-modal');
+export const hideModal = eventFactory('hide-modal');
+
+export const dsRequestResponse = eventFactory<DataSourceResponsePayload>('ds-request-response');
+export const dsRequestError = eventFactory<any>('ds-request-error');
+
+export const graphHover = eventFactory<GraphHoverPayload>('graph-hover');
+export const graphHoverClear = eventFactory('graph-hover-clear');
+
+export const toggleSidemenuMobile = eventFactory('toggle-sidemenu-mobile');
+export const toggleSidemenuHidden = eventFactory('toggle-sidemenu-hidden');
+
+export const playlistStarted = eventFactory('playlist-started');
+export const playlistStopped = eventFactory('playlist-stopped');
+
+export const toggleKioskMode = eventFactory<ToggleKioskModePayload>('toggle-kiosk-mode');
+export const toggleViewMode = eventFactory('toggle-view-mode');
+
+export const timeRangeUpdated = eventFactory<TimeRange>('time-range-updated');
+
+export const repeatsProcessed = eventFactory('repeats-processed');
+export const rowExpanded = eventFactory('row-expanded');
+export const rowCollapsed = eventFactory('row-collapsed');
+export const templateVariableValueUpdated = eventFactory('template-variable-value-updated');
+
+export const panelSizeChanged = eventFactory('panel-size-changed');
+export const panelInitialized = eventFactory('panel-initialized');
+export const panelTeardown = eventFactory('panel-teardown');
+export const panelChangeView = eventFactory<PanelChangeViewPayload>('panel-change-view');
+export const removePanel = eventFactory<RemovePanelPayload>('remove-panel');
+export const initPanelActions = eventFactory<MenuElement[]>('init-panel-actions');
+
+export const editModeInitialized = eventFactory('init-edit-mode');
+export const clientRefreshed = eventFactory('refresh');
+
+export const dataFramesReceived = eventFactory<DataFrame[]>('data-frames-received');
+
+export const componentDidMount = eventFactory('component-did-mount');
+
+export const graphClicked = eventFactory<GraphClickedPayload>('graph-click');
+
+export const thresholdChanged = eventFactory<ThresholdChangedPayload>('threshold-changed');
+
+export const zoomOut = eventFactory<number>('zoom-out');
+
+export const shiftTime = eventFactory<number>('shift-time');
+
+export const elasticQueryUpdated = eventFactory('elastic-query-updated');
+
+export const layoutModeChanged = eventFactory<string>('layout-mode-changed');
+
+export const jsonDiffReady = eventFactory('json-diff-ready');
+
+export const closeTimepicker = eventFactory('closeTimepicker');
+
+export const routeUpdated = eventFactory('$routeUpdate');
```

Comment:
Almost all of these are internal app events that we do not want to expose to plugins (and are legacy mechanics that we want to move away from over time). So most of these should be moved to app/types/events.ts (start with all until we know which are needed to be used from plugins).

Reply:
Sure, will do